text/0000-sbom.md

```diff
@@ -0,0 +1,155 @@
+
+## What use cases does it support?
+
+- Consistent SBOM across buildpacks
+- Improved integration with other supply chain analysis and scanning tools
+
+## What is the expected outcome?
+
+- Use of CycloneDX and SPDX as the SBOM storage format and updates the the specification and lifecycle to reflect the same.
+- Moving the SBOM from `<layer>.toml`, `launch.toml` and `build.toml` to separate files.
+
+## Why support multiple SBOM formats?
+
+The motivation for supporting multiple SBOM formats is because there is currently no clear leader in the SBOM space. SPDX and CycloneDX both specialize in different use cases - compliance and supply chain security respectively, and although both do have a lot of overlap, there is still loss of information when converting between the two. The other motivation is that the SBOM story doesn't end with their generation. Different users could have different SBOM consumers who support one format or the other. Leaving individual SBOM files intact allows users to post-process it later if needed without any loss in information. This allows users to flexibly use the Buildpacks API to generate a SBOM in the format that is most useful to them.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Currently the SBOM is defined in the `<layer>.toml`, `launch.toml` and `build.toml` respectively under the `[bom]` table.
+It may not be the most user-friendly way for buildpack authors to create SBOM documents in the above format. This RFC proposes that the `bom` be moved to `<layer>.bom.<ext>`, `launch.bom.<ext>` and `build.bom.<ext>` instead where `<ext>` will be `cdx.json` for CycloneDX documents and `spdx.json` for SPDX documents. A buildpack can also output the `bom` in multiple formats i.e. both CycloneDX and SPDX. 
+
+Additionally the buildpack must also indicate the SBOM format(s) it exports using the in its `buildapck.toml`'s `sbom` key in the `buildpacks` table.
+
+The format of this key should look like - 
+
+```toml
+api = "0.x"
+
+[buildpack]
+id = "<buildpack ID>"
+name = "<buildpack name>"
+# This can be an array of supported SBOM formats by the buildpack.
+# Valid array values are "CycloneDX" and "SPDX" as of this RFC.
+sbom = ["CycloneDX"]
+```
+
+This information can be displayed on the registry and `pack inspect-buildpack`.
+
+The lifecycle of the respective `bom` files during rebuilds would be exactly the same as the `bom` table currently. 
+
+The lifecycle would be responsible for taking all of the above `bom` files, and putting them in `/layers/config/sbom/<buildpack-id>/<layer-id>/<type>.bom.<ext>` for the layer `bom` files where `type` call be `build` or `launch` or `both`. Buildpack level BOM files will reside at `/layers/config/sbom/<buildpack-id>/build.bom.<ext>` and `/layers/config/sbom/<buildpack-id>/launch.bom.<ext>` for the `launch` and `build` `bom` respectively.
```

Comment:
Just to call it out explicitly, `type` would be determined by what is the value under the `[types]` table in the `<layer>.toml` file?

Reply:
Are we still putting bom information in `io.buildpacks.build.metadata`? If not, would it make sense for each `/layers/config/sbom/<buildpack-id>/<layer-id>/<type>.bom.<ext>` to be its own layer, to aid in layer re-use?

Could we translate this "[how it works](https://github.com/buildpacks/rfcs/blob/main/text/0087-bom-in-layer-metadata.md#how-it-works)" from the bom-in-layer-metadata RFC into this new world?

A couple of thoughts:
* If the layer is `cache = true` and there is a `<layer>.bom.<ext>` then the bom would be saved in the cache alongside the layer and restored to the buildpack's `<layers>` directory on the next build
* If the layer is `launch = true`, `cache = false`, `build = false` and there is a `<layer>.bom.<ext>` then normally we are only restoring the layer metadata, but would the buildpack also want the bom to be available, so that it could decide whether to re-use the layer or not? In this case we would need to copy the `/layers/config/sbom/<buildpack-id>/<layer-id>/<type>.bom.<ext>` from the previous image as well (unless this is going in a label?

Reply:
I feel like in the case of `launch = true`, `cache = false`, `build = false` I would want the `<layer>.bom.<ext>` or at least the information therein to be available so that I have the option to reuse the BOM and not generate a new one in the case that I am reusing that layer.

That is my two cents on the matter at least.

Reply:
This is the part that @ekcasey and I need to sync up about.